File: airportgap-java/src/test/java/spec/distance/DistanceBasicTest.java
package spec.distance;

import config.Env;
import http.ApiClient;
import org.junit.jupiter.api.Test;
import static io.restassured.RestAssured.given;
import static org.assertj.core.api.Assertions.assertThat;

public class DistanceBasicTest {
  private final ApiClient api = new ApiClient(Env.baseUrl(), Env.token());

  @Test
  void distanceBetweenKIXandNRT_hasUnits() {
    var body = "{\"from\":\"KIX\", \"to\":\"NRT\"}";
    var resp = given()
        .spec(api.baseSpec())
        .header("Content-Type", "application/json")
        .body(body)
        .post("/airports/distance")
        .then()
        .statusCode(200)
        .extract().jsonPath();

    Double miles = resp.getDouble("data.attributes.miles");
    Double km = resp.getDouble("data.attributes.kilometers");
    Double nm = resp.getDouble("data.attributes.nautical_miles");

    assertThat(miles).isNotNull().isGreaterThanOrEqualTo(0.0);
    assertThat(km).isNotNull().isGreaterThanOrEqualTo(0.0);
    assertThat(nm).isNotNull().isGreaterThanOrEqualTo(0.0);
  }
}

File: airportgap-typescript/tests/distance/distanceBasic.test.ts
import { describe, it, expect } from 'vitest'
import { ApiClient } from '../../src/http/apiClient.js'

describe('Distance basic', () => {
  const api = new ApiClient()

  it('KIX <-> NRT has unit values', async () => {
    const r = await api.base.post('/airports/distance', { from: 'KIX', to: 'NRT' })
    expect(r.status).toBe(200)
    const a = r.data?.data?.attributes
    expect(a.miles).toBeGreaterThanOrEqual(0)
    expect(a.kilometers).toBeGreaterThanOrEqual(0)
    expect(a.nautical_miles).toBeGreaterThanOrEqual(0)
  })
})
